keyvault/client/secrets/README.md

```diff
@@ -0,0 +1,337 @@
+- [Create a Secret](#create-a-secret)
+- [Retrieve a Secret](#retrieve-a-secret)
+- [Update an existing Secret](#update-an-existing-secret)
+- [Delete a Secret](#delete-a-secret)
+- [List Secrets](#list-secrets)
+
+### Create a Secret
+
+Create a Secret to be stored in the Azure Key Vault.
+- `setSecret` creates a new secret in the key vault. if the secret with name already exists then a new version of the secret is created.
+```Java
+SecretClient secretClient = SecretClient.builder()
+        .endpoint(<your-vault-url>)
+        .credential(new AzureCredential())
+        .build();
+
+Secret secret = secretClient.setSecret("secret_name", "secret_value").value();
+System.out.printf("Secret is created with name %s and value %s \n", secret.name(), secret.value());
+```
+
+### Retrieve a Secret
+
+Retrieve a previously stored Secret by calling `getSecret`.
+```Java
+SecretClient secretClient = SecretClient.builder()
+        .endpoint(<your-vault-url>)
+        .credential(new AzureCredential())
+        .build();
+
+Secret secret = secretClient.getSecret("secret_name").value();
+System.out.printf("Secret is returned with name %s and value %s \n", secret.name(), secret.value());
+```
+
+### Update an existing Secret
+
+Update an existing Secret by calling `updateSecret`.
+```Java
+SecretClient secretClient = SecretClient.builder()
+        .endpoint(<your-vault-url>)
+        .credential(new AzureCredential())
+        .build();
+
+// Get the secret to update.
+Secret secret = secretClient.getSecret("secret_name", "secret_value").value();
+// Update the expiry time of the secret.
+secret.expires(OffsetDateTime.now().plusDays(30));
+SecretBase updatedSecret = secretClient.updateSecret(secret).value();
+System.out.printf("Secret's updated expiry time %s \n", updatedSecret.expires().toString());
+```
+
+### Delete a Secret
+
+Delete an existing Secret by calling `deleteSecret`.
+```Java
+SecretClient secretClient = SecretClient.builder()
+        .endpoint(<your-vault-url>)
+        .credential(new AzureCredential())
+        .build();
+
+DeletedSecret deletedSecret = client.deleteSecret("secret_name").value();
+System.out.printf("Deleted Secret's deletion date %s", deletedSecret.deletedDate().toString());
+```
+
+### List Secrets
+
+List the secrets in the key vault by calling `listSecrets`.
+```Java
+SecretClient secretClient = SecretClient.builder()
+        .endpoint(<your-vault-url>)
+        .credential(new AzureCredential())
+        .build();
+        
+// The List Secrets operation returns secrets without their value, so for each secret returned we call `getSecret` to get its // value as well.
+secretClient.listSecrets().stream().map(secretClient::getSecret).forEach(secretResponse -> 
+  System.out.printf("Received secret with name %s and value %s", secretResponse.value().name(), secretResponse.value().value()));
```

Comment:
listSecrets for preview 1 should be returning an Iterable, not a List.

Reply:
Will update code, samples, tests and README for this in  (core-updates PR for KV).